File: 0043.MultiplyStrings.py
class Solution(object):
    def multiply(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        mul = [0] * (len(num1)+len(num2))
        pos = len(mul) - 1
        for d1 in reversed(num1):
            index = pos
            for d2 in reversed(num2):
                mul[index] += int(d1) * int(d2)
                mul[index-1] += mul[index]/10
                mul[index] %= 10
                index -= 1
            pos -= 1
        pos = 0
        while pos < len(mul)-1 and mul[pos] == 0:
            pos += 1
        ans = ''
        for i in xrange(pos, len(mul)):
            ans += str(mul[i])
        return ans

File: 0043.MultiplyStrings.go
func multiply(num1 string, num2 string) string {
	m, n := len(num1), len(num2)
	multi := make([]byte, m+n)
	for i := m - 1; i >= 0; i-- {
		for j := n - 1; j >= 0; j-- {
			multi[i+j+1] += (num1[i] - '0') * (num2[j] - '0')
			multi[i+j] += multi[i+j+1] / 10
			multi[i+j+1] %= 10
		}
	}
	idx := 0
	for ; idx < m+n-1 && multi[idx] == 0; idx++ {
	}
	for i := idx; i < m+n; i++ {
		multi[i] += '0'
	}
	return string(multi[idx:])
}